asreview/webapp/task_manager/task_manager.py

```diff
@@ -0,0 +1,258 @@
+                if action == "failure":
+                    logging.error(f"Failed to train model for project {project_id}")
+                self.remove_pending(project_id)
+
+    def _handle_incoming_messages(self, conn):
+        """Handles incoming traffic."""
+        client_buffer = ""
+        while True:
+            try:
+                data = conn.recv(self.receive_bytes)
+                if not data:
+                    # if client_buffer is full convert to json and
+                    # put in buffer
+                    if client_buffer != "":
+                        # we may be dealing with multiple messages,
+                        # update buffer to produce a correct json string
+                        client_buffer = "[" + client_buffer.replace("}{", "},{") + "]"
+                        messages = json.loads(client_buffer)
+                        logging.info(f"Received messages:\n{messages}\n")
+                        # add to buffer
+                        self.message_buffer.extend(deque(messages))
+                    # client disconnected
+                    break
+
+                else:
+                    message = data.decode("utf-8")
+                    client_buffer += message
+
+            except Exception:
+                break
+
+        conn.close()
+
+    def start_manager(self):
+        server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        server_socket.bind((self.host, self.port))
+        server_socket.listen()
+
+        # Set a timeout
+        server_socket.settimeout(0.1)
+
+        logging.info("...starting server")
+
+        while True:
+            try:
+                # Accept incoming connections with a timeout
+                conn, addr = server_socket.accept()
+                # Start a new thread to handle the client connection
+                client_thread = threading.Thread(
+                    target=self._handle_incoming_messages, args=(conn,)
+                )
+                client_thread.start()
+
+            except socket.timeout:
+                # No incoming connections => perform handling queue
+                self._process_buffer()
+                # Pop tasks from database into 'pending'
+                self.pop_waiting_queue()
+
+
+def setup_logging(verbose=False):
+    level = logging.INFO if verbose else logging.ERROR
+    logging.basicConfig(level=level, format="%(asctime)s - %(levelname)s - %(message)s")
+
+
+def run_task_manager(max_workers, host, port, verbose=False):
+    setup_logging(verbose)
+    manager = TaskManager(max_workers=max_workers, host=host, port=port)
+    manager.start_manager()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose logging")
+    args = parser.parse_args()
+
+    setup_logging(verbose=args.verbose)
+
+    manager = TaskManager(max_workers=2)
+    manager.start_manager()
```

Comment:
This is almost an entry point. It might be useful to make this an entry point. Users can have difficulty finding this file when installed via pip (and the location changes with an upgrade of Python). There is `entry_points_internal` if you don't want users to see this entry point via `asreview -h`.